app/scripts/modules/core/account/account.service.js

```diff
@@ -94,6 +94,42 @@ module.exports = angular.module('spinnaker.core.account.service', [
       return deferred.promise;
     });
 
+    let getUniqueRegionsForAllAccounts = _.memoize((provider) => {
+      return getRegionsKeyedByAccount(provider)
+        .then(function(regionsByAccount) {
+          let regions = _.chain(regionsByAccount)
+            .values()
+            .map(acct => acct.regions)
+            .flatten()
+            .map(reg => reg.name)
+            .uniq()
+            .value();
+
+          return regions;
+        });
+    });
+
+    let getUniqueGceZonesForAllAccounts = _.memoize((provider) => {
```

Comment:
@anotherchrisberry Saw your comment on the pending pr and agree. We should reconcile the differences between the different representations of regions/zones on an account. Took a quick look, and it looks doable. Going to be working through some of the zonal/regional bits on GCE starting next week or so. If you're ok leaving this method as-is until then, would like to tackle this at the same time.